src/nsls2api/cli/proposal.py

```diff
@@ -1,24 +1,173 @@
+                saf.saf_id, f"{status_icon} {saf.status}", instruments_str
+            )
+
+        console.print(Panel(saf_table, title="📎 Safety Forms", expand=True))
+
+    # Slack Channels 💬
+    if proposal.slack_channels:
+        slack_table = Table(expand=True)
+        slack_table.add_column("Channel Name", style="cyan")
+        slack_table.add_column("Channel ID", style="dim")
+
+        for sc in proposal.slack_channels:
+            slack_table.add_row(f"#{sc.channel_name}", sc.channel_id)
+
+        console.print(Panel(slack_table, title="💬 Slack Integration", expand=True))
+
+    # --- Metadata Panel ---
+    metadata_table = Table.grid(padding=(0, 2))
+    metadata_table.add_column(justify="right", style="dim")
+    metadata_table.add_column()
+
+    metadata_table.add_row(
+        "🕒 Created", proposal.created_on.strftime("%Y-%m-%d %H:%M:%S")
+    )
+    metadata_table.add_row(
+        "📝 Last Updated", proposal.last_updated.strftime("%Y-%m-%d %H:%M:%S")
+    )
+
+    console.print(
+        Panel(
+            metadata_table,
+            title="📌 Metadata",
+            style="dim",
+            expand=False,
+            box=box.ROUNDED,
+        )
+    )
 
 
 @app.command()
-def search(proposal: int):
-    print(f"Searching for Proposal: {proposal}")
-    with Progress(
-        SpinnerColumn(),
-        TextColumn("[progress.description]{task.description}"),
-        transient=True,
-    ) as progress:
-        progress.add_task(description="Searching...", total=None)
-        time.sleep(3)
-    print(f"Now showing all the awesome information about proposal {proposal}!")
+def view(
+    identifier: str, saf: bool = typer.Option(False, help="Treat identifier as SAF ID")
+):
+    lookup_type = "SAF ID" if saf else "Proposal ID"
+
+    if saf:
+        url = f"v1/proposal/saf/{identifier}"
+        error("Lookup by SAF ID is not yet implemented.")
+        raise typer.Exit(code=1)
+    else:
+        url = f"v1/proposal/{identifier}"
+
+    response = call_nsls2api_endpoint(url, method="GET")
+
+    if response is None:
+        error(f"Failed to retrieve proposal with {lookup_type} = {identifier}.")
+        raise typer.Exit(code=1)
+
+    proposal_data = response.json().get("proposal", {})
+    if not proposal_data:
+        error(f"No data found for proposal with {lookup_type} = {identifier}.")
+        raise typer.Exit(code=1)
+
+    # Assuming Proposal is a Pydantic model
+    proposal = ProposalDisplay(**proposal_data)
+    display_proposal(proposal)
+
+
+# @app.command()
+# def search(proposal: int):
+#     print(f"Searching for Proposal: {proposal}")
+#     with Progress(
+#         SpinnerColumn(),
+#         TextColumn("[progress.description]{task.description}"),
+#         transient=True,
+#     ) as progress:
+#         progress.add_task(description="Searching...", total=None)
+#         time.sleep(3)
+#     print(f"Now showing all the awesome information about proposal {proposal}!")
```

Comment:
Could be removed from this commit.

Reply:
Indeed it can - it can be added back in sometime in the future.